### PR TITLE
[`Awq`] Enable the possibility to skip quantization for some target modules

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Adds the possibility to load AWQ models if some modules of the model are skipped for quantization. 
E.g. for whisper, Llava, Mixtral, we respectively don't want to quantize the encoder, vision encoder and the gate layer to ensure inference stability. 

Let's merge it once AWQ makes the 0.1.8 release

cc @ArthurZucker @casper-hansen @TheBloke @SunMarc 

https://github.com/casper-hansen/AutoAWQ/pull/248

This PR makes it also possible to run multi-modal models with AWQ:

```py
from transformers import pipeline
from PIL import Image    
import requests

model_id = "ybelkada/llava-1.5-7b-hf-awq"
pipe = pipeline("image-to-text", model=model_id, device=0)
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/compel-neg.png"

image = Image.open(requests.get(url, stream=True).raw)
prompt = "USER: <image>\nCan you please describe this image?\nASSISTANT:"

outputs = pipe(image, prompt=prompt, generate_kwargs={"max_new_tokens": 100})
print(outputs[0]["generated_text"])
```

image

> USER:  \nCan you please describe this image?\nASSISTANT: The image features a brown and white cat sitting on a green surface, possibly a carpet or a grassy area. The cat is holding a red ball in its paws, seemingly playing with it. The cat appears to be focused on the ball, possibly preparing to play or just enjoying the toy.

### CODE REVIEW

**Comment by younesbelkada** [comment]
This PR enables compatiblity with Mixtral AWQ ! 
https://github.com/casper-hansen/AutoAWQ/pull/251 being merged, this PR is ready for review 🙏

**Review by amyeroberts — COMMENTED** [review]
Very nice! 🔥 

I just have a quick q on the test. Otherwise LGTM

File: tests/quantization/autoawq/test_awq.py [hunk 1]
```diff
@@ -223,6 +224,24 @@ def test_quantized_model_multi_gpu(self):
...
+    def test_quantized_model_no_k_proj_quantized(self):
+        """
+        Simple test that checks if the quantized model is working properly with multiple GPUs
+        """
+        dummy_input = torch.LongTensor([[0, 1, 0]]).to(torch_device)
+
+        quantized_model = AutoModelForCausalLM.from_pretrained(self.model_with_no_k_proj_quantized).to(torch_device)
+
+        self.assertTrue(isinstance(quantized_model.model.decoder.layers[0].self_attn.k_proj, torch.nn.Linear))
+        self.assertFalse(isinstance(quantized_model.model.decoder.layers[0].self_attn.k_proj, torch.nn.Linear))
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply tests/quantization/autoawq/test_awq.py]: Ah nice catch yes! Actually `v_proj` should be a quantized linear (hence not a nn.Linear), changed the test a bit!

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this capability!

**Comment by younesbelkada** [comment]
Thanks a lot for the review @amyeroberts ! 
I will merge this as soon as AWQ makes the 0.1.8 Mixtral release cc @casper-hansen just for your information

**Comment by younesbelkada** [comment]
Release has been done! Merging!

**Pull request merged**